File: backend/SharingService/Models/ShareToken.cs
namespace SharingService.Models;

public class ShareToken
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TravelPlanId { get; set; }
    public string Token { get; set; } = string.Empty;
    public string AccessType { get; set; } = "VIEW";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? ExpiresAt { get; set; }
}

File: frontend/src/models/ShareToken.js
export class ShareToken {
  constructor(data = {}) {
    this.id = data.id || '';
    this.travelPlanId = data.travelPlanId || '';
    this.token = data.token || '';
    this.accessType = data.accessType || 'VIEW';
    this.qrCodeBase64 = data.qrCodeBase64 || '';
    this.createdAt = data.createdAt || null;
    this.expiresAt = data.expiresAt || null;
  }
}
